public/src/components/channelManagement/epicTests/epicTestVariantEditor.tsx

```diff
@@ -101,33 +107,43 @@ const EpicTestVariantEditor: React.FC<EpicTestVariantEditorProps> = ({
   useEffect(() => {
     const isValid = Object.keys(errors).length === 0;
     onValidationChange(isValid);
-  }, [
```

Comment:
with this change we've lost validation, e.g. I can now save an invalid template in the text. I'll have a think about how we can make the new component work with the existing validation

Reply:
For context, validating the copy in the RTE (for either zero length or excessive length) caused me endless headaches because the current code uses react-hook-form for this work. Sadly RHF expects the field it is checking to be some sort of input element (with a .value attribute, I think) whereas ProseMirror avoids <input> and <textarea> and uses contetEditable <div> elements instead. I'll check to make sure I've correctly plumbed in the replacement validation code - for four of those five fields - into this useEffect hook.

Reply:
react-hook-form provides a way to handle custom inputs - https://react-hook-form.com/api/usecontroller/controller#main
If we are to continue using react-hook-form in this codebase then we should wrap the new input with this.

I've just had a play and this works for `paragraphs` (note the use of `templateValidator`):
```
troller
  name="paragraphs"
  control={control}
  rules={{
    required: true,
    validate: (pars: string[]) => pars.map(templateValidator).find(result => result !== true),
  }}
  render={(data) => {
    return <RichTextEditor
      error={errors.paragraphs !== undefined}
      // @ts-ignore -- react-hook-form doesn't believe it has a message field
      helperText={errors?.paragraphs?.message || getParagraphsHelperText()}
      copyData={data.value}
      updateCopy={pars => {
        data.onChange(pars);
        handleSubmit(onSubmit)()
      }}
      name="paragraphs"
      label="Body copy"
      disabled={!editMode}
    />
  }}
/>
```

Happy to pair on this - it's quite fiddly

Reply:
I think I've implemented this correctly for `epicTestVariantEditor.tsx` - needs more testing. Also need to implement this fix for `bannerTestVariantEditor.tsx`